### PR TITLE
fix(tenant-selector): remove confusing "x" on invalid tenant

### PR DESCRIPTION
**Pull request by stonecrafter**
When the tenant is invalid, the UI displays an invalid state with
- an error message
- the input box highlighted in red
- a red X inside the input box

This is antd's default behaviour but it is confusing for users that there is an X inside the input field, which does nothing. We may expect that it should be a clickable button to clear out the invalid input.

Would like to just remove it to avoid confusion.

Fixes [OPSUP-388]

[OPSUP-388]: https://cognitedata.atlassian.net/browse/OPSUP-388

### CODE REVIEW

**Comment by stonecrafter** [comment]
Example

Screenshot 2020-02-03 at 10 32 33

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-623.gearboxjs.preview.cogniteapp.com

**Review by rumeshwick — COMMENTED** [review]
@stonecrafter Agreed with you on this 'X icon' confusion and this change is looks good to me.

**Review by cemalbektas — APPROVED** [review]

**Comment by mrtraser** [comment]
Hey @stonecrafter, we can achieve results that you've described, but just within another way. Could you please expand `TenantSelectorProps` with `inputBehaviorOptions`:
```typescript
import { FormItemProps } from 'antd/lib/form/FormItem';

interface TenantSelectorProps {
  // ...
  inputBehaviorOptions: Pick<FormItemProps, 'extra' | 'hasFeedback'>,
}
```

and defaultProp for it will be:

```typescript
inputBehaviorOptions: {
  hasFeedback: true,
}
```
Then you can just pass `{ ...inputBehaviorOptions }` to the `Form.Item`.

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-623.gearboxjs.preview.cogniteapp.com

File: src/components/TenantSelector/stories/basic.story.mdx [hunk 1]
```diff
@@ -171,6 +172,26 @@ const validateTenantFailed = async (tenant: string) => {
   </Story>
 </Preview>
 
+### Custom input behavior options
+
+```typescript
+const inputBehaviorOptions = {
+  hasFeedback: false,
```
Comment (polomani) [inline src/components/TenantSelector/stories/basic.story.mdx]: could you use `extra` prop here as well? so it's easier to see what `inputBehaviorOptions` can also do

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-623.gearboxjs.preview.cogniteapp.com

File: src/components/TenantSelector/TenantSelector.tsx [hunk 2]
```diff
@@ -117,7 +121,7 @@ class TenantSelector extends Component<
           </SubTitle>
         )}
         <Form onSubmit={this.checkTenantValidity}>
-          <Form.Item hasFeedback={true} {...formItemProps}>
+          <Form.Item {...inputBehaviorOptions} {...formItemProps}>
```
Comment (mrtraser) [inline src/components/TenantSelector/TenantSelector.tsx]: You can put it inside single curly brackets, like:
```jsx
<Form.Item {...inputBehaviorOptions, ...formItemProps}>
```

Comment (stonecrafter) [reply src/components/TenantSelector/TenantSelector.tsx]: [I don't think it works?](https://stackoverflow.com/questions/49470037/how-to-pass-several-props-in-spread-operator-in-react)

File: src/components/TenantSelector/stories/basic.story.mdx (hunk 1 above)
Comment (stonecrafter) [reply src/components/TenantSelector/stories/basic.story.mdx]: I don't think there is a use for `extra` prop. I just copied it directly from above. Will remove it

File: src/components/TenantSelector/TenantSelector.tsx (hunk 2 above)
Comment (mrtraser) [reply src/components/TenantSelector/TenantSelector.tsx]: yep, you're right, then I'm ok with your solution

File: src/components/TenantSelector/stories/basic.story.mdx (hunk 1 above)
Comment (mrtraser) [reply src/components/TenantSelector/stories/basic.story.mdx]: No, we haven't used it, but it was like an improvement to input configuration. It comes from [antd](https://ant.design/components/form/?locale=en-US#Form.Item)

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-623.gearboxjs.preview.cogniteapp.com

File: src/components/TenantSelector/stories/basic.story.mdx (hunk 1 above)
Comment (stonecrafter) [reply src/components/TenantSelector/stories/basic.story.mdx]: I see! I understand now

**Comment by cognite-cicd** [comment]
This PR's storybook can now be previewed at the following URL: https://pr-623.gearboxjs.preview.cogniteapp.com

**Review by mrtraser — APPROVED** [review]
LGTM 👍

**Comment by stonecrafter** [comment]
When will be the next release that includes this change? :)

**Comment by polomani** [comment]
@stonecrafter it will, travis stuck because stupid dependabot created like ~20 prs on 3d-reveal repo

**Pull request merged**